File: DR/1_21/10815.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int search(int arr[], int length, int target) { // 반복적 
	int start = 0;
	int end = length-1;
	int mid;
	
	while(start<=end) {
		mid = (start+end)/2;
		if(arr[mid] == target) {
			return mid;
		} else {
			if(arr[mid] > target) {
				end = mid-1;
			} else {
				start = mid+1;
			}
		}
	}
	return -1;
}

bool cmp(int a, int b) {
	return a<b;
}

int main()
{							
	ios::sync_with_stdio(false);
	cin.tie(0);	 
	int n, m, result;
	cin>>n;
	
	int store[n];
	for(int i = 0; i<n; i++) {
		cin>>store[i];
	}
	sort(store, store+n, cmp);
	
	cin>>m;
	int custom[m];
	for(int i = 0; i<m; i++) {
		cin>>custom[i];
	}
	
	for(int i = 0; i<m; i++) { 
		result = search(store, n, custom[i]);
		if(result==-1) {
			cout<<0<<' ';
		} else {
			cout<<1<<' ';
		}
	}
	return 0;
}

File: YR/0211/10815.py
#10815
#실버5

#범위가 매우큼 -> 이분탐색 이용

#간단한 문제지만 이분탐색을 어떻게 이용해야 할지 모르겠군..
#요소들을 확인하는 문제는 이진탐색.
n=int(input())
arr=[]
check=[]

arr=list(map(int,input().split()))    
m=int(input())

check=list(map(int,input().split())) 

arr.sort()
    
def search(c,start,end):
    while(start<=end):
        mid=(start+end)//2
        
        if c>arr[mid]:
            start=mid+1
        elif c==arr[mid]:
            return 1
        else:
            end=mid-1
        
    return 0    
    
for c in check:
    if not search(c,0,n-1):
        print(0,end=' ')
    else:
        print(1,end=' ')
